tsdb/querier.go

```diff
@@ -387,39 +387,59 @@ func postingsForMatcher(ix IndexReader, m *labels.Matcher) (index.Postings, erro
 		}
 	}
 
-	vals, err := ix.LabelValues(m.Name)
+	vals, err := ix.LabelValuesUnsorted(m.Name)
 	if err != nil {
 		return nil, err
 	}
 
 	var res []string
+	lastVal, isUnsorted := "", false
 	for _, val := range vals {
 		if m.Matches(val) {
 			res = append(res, val)
+			if val < lastVal {
```

Comment:
No need to check "val < lastVal" if `isUnsorted` is already `true`
```suggestion
			if !isUnsorted && val < lastVal {
```

_Same comment applies to `inversePostingsForMatcher()`

Reply:
Removed the check entirely, as the sort call would check this anyway and would be quick for a sorted slice.